presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java

```diff
@@ -1386,6 +1409,11 @@ public void testArrayRemove()
                 new ArrayType(createDecimalType(22, 10)),
                 ImmutableList.of(decimal("1234567890.1234567890"), decimal("9876543210.9876543210"), decimal("123123123456.6549876543")));
         assertCachedInstanceHasBoundedRetainedSize("ARRAY_REMOVE(ARRAY ['foo', 'bar', 'baz'], 'foo')");
+
+        assertFunction("ARRAY_REMOVE(ARRAY [ARRAY[CAST(null AS BIGINT)]], ARRAY[CAST(1 AS BIGINT)])", new ArrayType(new ArrayType(BIGINT)), ImmutableList.of(singletonList(null)));
+        // TODO: With current semantics we cannot remove a null element, similarly to ARRAY_REMOVE(ARRAY [NULL], NULL) => ARRAY [NULL]
+        // TODO: It seems we should apply IS_DISTINCT_FROM semantics
+        assertFunction("ARRAY_REMOVE(ARRAY [ARRAY[CAST(null AS BIGINT)]], ARRAY[CAST(null AS BIGINT)])", new ArrayType(new ArrayType(BIGINT)), ImmutableList.of(singletonList(null)));
```

Comment:
We need to decide on the semantics before merging this PR.

Now that we specified array equal semantics for null elements in structural types, I think the semantics here need to be specified. Previously, we delayed making a decision on these issues by simply failing these queries. But now that these queries will return a result, we have make a choice now. We want to avoid having to change the semantics in the future.

Reply:
I'm following the semantics. Currently we remove the element, if, and only if the comparison returns `TRUE`. We do not remove `NULL` even if there is `NULL` in the array.

@martint